### PR TITLE
React carousel component - Paolo Di Martino

### PR DESCRIPTION
**Pull request by Paolo-DM**
Kapture 2022-05-24 at 01 17 36

### CODE REVIEW

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking great, just one update needed. The way you set props to a property of `this` will work fine for this exercise but it is bad habit to get into, it will lead to issues with updates down the road. Let me know if you have any questions.

File: react-accordion-component/src/accordion.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.topics = props.topics;
```
[the remark this answers is not in the document]
Comment (Paolo-DM) [reply react-accordion-component/src/accordion.jsx:5]: Thanks for the advice! I've modified the code in the Carousel component (I don't know why but in this pull request there was also the commit from the Accordion exercise)

[the remark this answers is not in the document]
Comment (scbowler) [reply react-accordion-component/src/accordion.jsx:5]: Oh oops! Sorry about that didn't even notice this wasn't the PR for Accordion.

File: react-carousel-component/src/carousel.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    clearInterval(this.intervalId);
+    this.intervalId = setInterval(this.rightClick, 3000);
+  }
+
+  renderDots() {
+    const out = [];
+    for (let i = 0; i < this.props.images.length; i++) {
+      out.push(<i key={this.props.images[i].id} className={this.state.selectedImg === i + 1 ? 'ph-circle-fill' : 'ph-circle-bold'} onClick={() => this.handleClick(i + 1)} ></i>);
+    }
+    return out;
```
Comment (scbowler) [inline react-carousel-component/src/carousel.jsx:29]: Here it is better and more in line with "modern" practices to use `.map`

**Review by scbowler — APPROVED** [review]
Carousel looks great! 👍 

1 FYI comment, no changes required.

**Pull request merged**